Remove parameter "options = nil" for #clear

ysksn opened the pull request:
`options = nil` never been needed for `#clear`.

rafaelfranca commented:
Thank you for the pull request. It is there to be used by other backends that needs options. It is an API not necessarily needs to be used in our backends.

rafaelfranca commented:
I'm fine with doing this to Rails 5.1 but we need a changelog entry. Could you add one?

ysksn committed:
Update CHANGELOG.md for #25628 [ci skip]

Move new CHANGELOG entry top [ci skip]

Remove parameter "options = nil" for #clear

prathamesh-sonpatki commented on activesupport/CHANGELOG.md:
```diff
@@ -76,4 +76,10 @@
+    `ActiveSupport::Cache::Strategy::LocalCache::LocalStore` and
+    `ActiveSupport::Cache::Strategy::LocalCache`.
+
+    *Yosuke Kabuto*
```
Please add the new CHANGELOG entry to the top of the file.

prathamesh-sonpatki commented:
@ysksn Please also squash the commits, thanks!

maclover7 commented:
Hmm, this kind of puts `#clear` in a weird place -- would it be better to maintain consistency across all cache store operation methods, and allow an `options` hash to be passed through? 😬

rafaelfranca commented:
> Hmm, this kind of puts #clear in a weird place -- would it be better to maintain consistency across all cache store operation methods, and allow an options hash to be passed through?

Accepting this argument and not doing anything already caused problems in the applications I worked, people though that `clear` did the same as `delete` and passed argument to delete only one cache entry. As nothing raised the whole cache was being cleaned.

We are not using the argument and if a backend uses it can define it.

jeremy commented:
Reverted (816a3763d9bb070bf339de32c6a3ce31a45369f8) and backported to 5-1-stable (04b6e71d8f45a5436da8206a9e09ea0017091111).

Rationale: https://github.com/rails/rails/issues/29067#issuecomment-301342084
